deepinfra/models/base/base.py

```diff
@@ -1,8 +1,38 @@
+import os
+
 from deepinfra.clients import DeepInfraClient
+from deepinfra.constants.client import ROOT_URL
+from deepinfra.utils.url import URLUtils
 
 
 class BaseModel:
-    def __init__(self, endpoint, auth_token):
-        self.endpoint = endpoint
-        self.auth_token = auth_token
-        self.client = DeepInfraClient(self.endpoint, self.auth_token)
\ No newline at end of file
+	"""
+	Base class for all models
+	@param endpoint: The endpoint of the model or the model name.
+	@param auth_token: The API key to authenticate the requests. If not provided, it will be fetched from the environment.
+	"""
+	def __init__(self, endpoint, auth_token:str = None):
+		if URLUtils.is_valid_url(endpoint):
+			self.endpoint = endpoint
+		else:
+			self.endpoint = ROOT_URL + endpoint
+		self.auth_token = auth_token or self._get_auth_token_from_env() or self._warn_about_missing_api_key()
+		self.client = DeepInfraClient(self.endpoint, self.auth_token)
+
+	def _get_auth_token_from_env(self):
+		"""
```

Comment:
Maybe in another commit, but replace all tabs with 4 spaces.

Reply:
Noted